design/proposals/authenticator_management_improvement.md

```diff
@@ -0,0 +1,266 @@
+      - [Functionality](#functionality)
+      - [Synopsis](#synopsis)
+      - [Usage Example](#usage-example)
+      - [API Specification](#api-specification)
+    - [Delete Authenticator](#delete-authenticator)
+      - [Functionality](#functionality-1)
+      - [Synopsis](#synopsis-1)
+      - [Usage Example](#usage-example-1)
+      - [API Specification](#api-specification-1)
+    - [Update Authenticator](#update-authenticator)
+      - [Functionality](#functionality-2)
+      - [Synopsis](#synopsis-2)
+      - [Usage Example](#usage-example-2)
+      - [API Specification](#api-specification-2)
+    - [Enable Authenticator](#enable-authenticator)
+      - [Functionality](#functionality-3)
+      - [Synopsis](#synopsis-3)
+      - [Usage Example](#usage-example-3)
+      - [API Specification](#api-specification-3)
+    - [Disable Authenticator](#disable-authenticator)
+      - [Functionality](#functionality-4)
+      - [Synopsis](#synopsis-4)
+      - [Usage Example](#usage-example-4)
+      - [API Specification](#api-specification-4)
+  - [Implementation Details](#implementation-details)
+  - [Backwards Compatibility](#backwards-compatibility)
+  - [Performance](#performance)
+  - [Security](#security)
+  - [Documentation](#documentation)
+  - [Version Update](#version-update)
+  - [Delivery Plan](#delivery-plan)
+  
+## Introduction
+
+Conjur authenticators are defined as sub-policies, under the `conjur` policy.  
+Currently to define an authenticator, the user needs to complete the following steps:
+
+- Read the documentation
+- Understand how the authenticator policy is structured
+- Create a policy yaml file that defines the authenticator
+- Load the policy using Conjur CLI
+- Fill the authenticator policy variables with their values
+- Enable the authenticator through an environment variable or REST API call
+
+To delete an authenticator, the user needs to complete these steps.  
+Either:
+
+- Create a policy with a `!delete` statement, on the authenticator policy
+- Load the policy using Conjur CLI
+
+Or:
+
+- Update the `conjur` policy yaml file and take out the authenticator
+- Reload the `conjur` policy using the `--replace` option  
+
+The current user experience requires a certain level of expertise. The user needs to understand the policy structure of each authenticator and the process is comprised of multiple steps. This proposal offers an easier way for managing authenticators.
+
+## Proposed Solution
+
+The proposed solution is to add a few simple Conjur CLI commands (and possibly APIs, see two alternatives below), that would hide the complexity and provide a single step to perform each action:
+
+### Add Authenticator
+
+#### Functionality
+
+Defines a new authenticator in Conjur. Behind the scenes, this command will creat the authenticator policy and fill it with all its required data. If the data is not required to be given by the user, like the `authn-k8s` CA cert and key, the information will be generated automatically.
```

Comment:
@micahlee has a prototype/POC with this functionality floating around.  It came up with the OCP Follower deployment improvements spike.  It's pretty powerful stuff.